### ISSUE TITLE
Feature request: allow user to disable autocomplete on emoji

### ISSUE THREAD

**Issue by rndmh3ro** [RocketChat/feature-requests#415]
### Description:

We're facing the problem described in RocketChat/Rocket.Chat#2469: If you enter an emoji at the end of the line, you have to press enter twice. The first time to autocomplete the emoji, the second time to send the message.

There are at least two ways to fix this:
* allow the user to disable autocomplete for emojies
* do not try to autocomplete if a complete emoji is already entered (like `:ok:` )

I guess both options should be implemented since the first one is basically just a workaround.

### Server Setup Information:

* Version of Rocket.Chat Server: 0.58.2
* Operating System: ubuntu
* Deployment Method(snap/docker/tar/etc): snap
* Number of Running Instances:  1
* DB Replicaset Oplog: yes

**Comment by Arnall** [comment]
Hi everyone, 
same problem here, i understand that people using english don't have the problem because the `:` is stuck to the previous word and so no list of emoji appears. But with french language for example it's a real nightmare to use `:` in RocketChat. It's very disturbing to have a popup list each time, and if you just want a newline after the `:` it's always finished with an emoji instead...really frustrating.
Plz add an option to disable this behavior (perhaps link it to the existing option : 'convert ascii to emoji'), or as @Crocmagnon wrote the list should wait for a minimum number of characters after the `:`

**Comment by tmartincpp** [comment]
To suit our needs, we replaced the ':' trigger by '::' in the source code : 
```
--- Rocket.Chat-3.15.0.orig/app/ui-message/client/popup/messagePopupConfig.js   2021-05-28 06:57:45.000000000 +0200
+++ Rocket.Chat-3.15.0/app/ui-message/client/popup/messagePopupConfig.js        2021-06-02 12:41:02.308525403 +0200
@@ -445,7 +445,7 @@ Template.messagePopupConfig.helpers({
                        title: t('Emoji'),
                        collection: emoji.list,
                        template: 'messagePopupEmoji',
-                       trigger: ':',
+                       trigger: '::',
                        prefix: '',
                        suffix: ' ',
                        rid: this.rid,
```

A config parameter to define this trigger would be awesome IMO.

**Comment by exec228** [comment]
`7) 8) 9) to 7) 😎 9) ?`
You are kidding.
Some of templates of Rocket Chat are really annoying.

**Comment by JakubJachym** [comment]
Yeah, still annoying. You type `:D` and press enter to send message. You end up with `:england:`.

Makes soooo much sense. Thanks! I am at a point where I want to disable completely all autocomplete as it's totally crazy all the time. You type `@alex`, in the middle you press tab and get `@lois`. Just because it thinks it makes sense. NO IT DOES NOT! 👎🏻

**Comment by Redfoxbox** [comment]
Please turn it off. Very annoying.

**Comment by exec228** [comment]
I'd rather remap smileys/emoji to rarest key-strings.
Anatomically «8 )» must at least have a nose «8-)» !
Which is exactly «the rare character occasion to avoid false positives».

If it be still dangerous for some of alternative-syntax languages, make it even farther rare.
Like /spectacles/

If remapping is not possible, I am qualified enough to read smileys from plain text.

**Comment by ValheruPL** [comment]
Any update ? ://

**Comment by Redfoxbox** [comment]
![image](https://github.com/RocketChat/feature-requests/assets/25455942/2ed755fc-cf89-477b-93bd-c89e706b7d37)
I just want to use my ":D" smiley and autocomplete wont allow me to do.

**Comment by kspes** [comment]
Man I hate this issue with a passion! Why can't this be resolved already?

**Comment by wullxz** [comment]
It looks like there is a patch for this in Version 6.3.0 that disables the suggestions if emojis are disabled.
Patch: https://github.com/RocketChat/Rocket.Chat/commit/9ea8088f0621900fa7a11156a89f7447482e4df8
Version: https://github.com/RocketChat/Rocket.Chat/releases/tag/6.3.0 (search for "emoji" - it's one of the first couple of changes mentioned)

**Comment by kspes** [comment]
Confirmed, updated docker container from 6.2.0 to 6.3.1 and the dropdown is no more!

**Comment by MartinSchoeler** [comment]
Closing this as issue seems to be resolved

**Comment by exec228** [comment]
Oh, really?
Where an End User can find such an option in `https://github.com/RocketChat/Rocket.Chat.Electron`, which is insistently recommended by RocketChat team?
I have 3.9.9, and I cannot find this feature ¯\\\_(ツ)\_/¯

**Comment by adminas8dev** [comment]
> Oh, really? Where an End User can find such an option in `https://github.com/RocketChat/Rocket.Chat.Electron`, which is insistently recommended by RocketChat team? I have 3.9.9, and I cannot find this feature ¯\_(ツ)_/¯

This "feature" was fixed by the RC server upgrade to version of 6.3.1, the client's update does not help.

**Comment by JakubJachym** [comment]
@exec228 : IIRC they _fixed it_ by triggering autocomplete on 3+ chars long strings if they start with colon, so if you type `:D` it won't trigger autocomplete. So technically the subject of issue was fixed. Also autocomplete of usernames seems to be more sane. 

But yea, we can't decide if we want to enable or disable emoji autocomplete. But for me, the bug is gone and I am happy with it.
